profile/filter_test.go

```diff
@@ -15,60 +15,398 @@
 package profile
 
 import (
+	"fmt"
 	"regexp"
+	"strings"
 	"testing"
+
+	"github.com/google/pprof/internal/proftest"
 )
 
-func TestFilter(t *testing.T) {
-	// Perform several forms of filtering on the test profile.
+var simpleProfileMappings = []*Mapping{
```

Comment:
Also, curious - why separate profiles with inlines and no inlines? These could be just separate samples in the same profile - some that have multi-line locations and some that don't?

Reply:
couple reasons:
- these test cases are largely duplicated from the cloud profiler test cases, so i didnt have to design them from scratch, and i dont have to worry missing an edge case that would lead to incompatibility.
- i think smaller test input are easier to understand and reason about.
- i think smaller test inputs are also easier to maintain, because the test cases dont all depend on the same profile.  if we used one profile for everything, then any modification to it would require changing all test cases. with multiple small profiles its easier to create new tests without breaking old ones.